### PR TITLE
Add mantapay to manta runtime

### LINKED ISSUE #1131: Configure mantapay for manta runtime
- [x] configuration
- [x] related tests, like rpc client tests, performance tests.
- [x] benchmark

### PR DESCRIPTION
**Pull request by Dengjianping**
## Description

- mantapay rpc performance tests and correctness tests will be added by another pr

---

Before we can approve this PR for merge, please make sure that **all** the following items have been checked off:
- [ ] Connected to an issue with discussion and accepted design using zenhub "Connect issue" button below
- [x] Added **one** label out of the `L-` group to this PR
- [x] Added **one or more** labels from the `A-` and `C-` groups to this PR
- [x] Explicitly labelled `A-calamari` and/or `A-manta` if your changes are meant for/impact either of these (CI depends on it)
- [ ] Re-reviewed `Files changed` in the Github PR explorer.

Situational Notes:
- If adding functionality, write unit tests!
- If importing a new pallet, choose a proper module index for it, and allow it in `BaseFilter`. Ensure **every** extrinsic works from front-end. If there's corresponding tool, ensure both work for each other.
- If needed, update our Javascript/Typescript APIs. These APIs are officially used by exchanges or community developers.
- If modifying existing runtime storage items, make sure to implement storage migrations for the runtime and test them with `try-runtime`. This includes migrations inherited from upstream changes, and you can search the diffs for modifications of `#[pallet::storage]` items to check for any.

### CODE REVIEW

File: node/src/command.rs [hunk 1]
```diff
@@ -458,6 +458,12 @@ pub fn run_with(cli: Cli) -> Result {
                             rpc::create_calamari_full,
                         ).await
                             .map_err(Into::into);
+                    } else if config.chain_spec.is_calamari() {
```
Comment (ghzlatarev) [inline node/src/command.rs]: is_manta()

File: node/src/rpc/mod.rs [hunk 2]
```diff
@@ -26,12 +26,12 @@ use sp_blockchain::{Error as BlockChainError, HeaderBackend, HeaderMetadata};
...
 
 mod calamari;
-mod common;
 mod dolphin;
+mod manta;
 
 pub use calamari::create_calamari_full;
-pub use common::create_common_full;
 pub use dolphin::create_dolphin_full;
+pub use manta::create_manta_full;
```
Comment (ferrell-code) [inline node/src/rpc/mod.rs]:
```suggestion
pub use self::manta::create_manta_full;
```

It will give you a weird error about ambiguous naming

**Review by ferrell-code — COMMENTED** [review]
looks good, one comment

**Review by ghzlatarev — COMMENTED** [review]
Looking good, please runt he SDK examples once against a local node just to make sure everything works (in case you havent')

**Comment by ghzlatarev** [comment]
nice one @ferrell-code

**Comment by Dengjianping** [comment]
> Looking good, please runt he SDK examples once against a local node just to make sure everything works (in case you havent')

It' done with Danial's help.

image

**Comment by ghzlatarev** [comment]
> > Looking good, please runt he SDK examples once against a local node just to make sure everything works (in case you havent')
> 
> It' done with Danial's help.
> 
> image

Let's hope the ci works with c5.9xlarge ...

**Review by ghzlatarev — APPROVED** [review]

**Review by ghzlatarev — DISMISSED** [review]

**Review by ferrell-code — APPROVED** [review]

**Review by ghzlatarev — APPROVED** [review]

**Pull request merged**